src/main/java/com/werfen/report/service/GridReportService.java

```diff
@@ -23,6 +24,14 @@ public File build(GridReportConfiguration gridReportConfiguration, GridPageDataS
         }
     }
 
+    public ByteArrayOutputStream buildToStream(GridReportConfiguration gridReportConfiguration, GridPageDataSource gridPageDataSource, ReportFormat reportFormat, PageFormat pageFormat) throws ReportException, ReportFormatException {
```

Comment:
I would seriously consider modifying the library interface (even if that implies modifying major version number) to have in all methods an indication of the type of the returned value. Thus, I would suggest having the following public methods:
* buildFile
* buildStream (without the unnecessary 'to')

This change should also apply to naming of private methods:
* buildPDFFile
* buildPDFStream
* buildExcelFile
* buildExcelStream

Reply:
Agree, all will be renamed according to your suggestions in the following commit.